charts/jenkins/values.yaml

```diff
@@ -82,274 +114,366 @@ controller:
+  servicePort: 8080
+  # -- k8s target port
+  targetPort: 8080
+  # -- k8s node port. Only used if serviceType is NodePort
+  nodePort:
+  # Use Local to preserve the client source IP and avoids a second hop for LoadBalancer and NodePort type services,
   # but risks potentially imbalanced traffic spreading.
   serviceExternalTrafficPolicy:
-  # Jenkins controller service annotations
+  # -- Jenkins controller service annotations
   serviceAnnotations: {}
-  # Jenkins controller custom labels
+  # -- Jenkins controller custom labels for the StatefulSet
   statefulSetLabels: {}
   #   foo: bar
   #   bar: foo
-  # Jenkins controller service labels
+  # -- Labels for the Jenkins controller-service
   serviceLabels: {}
   #   service.beta.kubernetes.io/aws-load-balancer-backend-protocol: https
+  # -- Custom Pod labels (an object with `label-key: label-value` pairs)
   # Put labels on Jenkins controller pod
   podLabels: {}
-  # Used to create Ingress record (should be used with ServiceType: ClusterIP)
-  # nodePort: <to set explicitly, choose port between 30000-32767
+
+  # -- Enable Kubernetes Probes configuration configured in `controller.probes`
   # Enable Kubernetes Startup, Liveness and Readiness Probes
   # if Startup Probe is supported, enable it too
   # ~ 2 minutes to allow Jenkins to restart when upgrading plugins. Set ReadinessTimeout to be shorter than LivenessTimeout.
+  # ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes
   healthProbes: true
   probes:
     startupProbe:
+      # -- Set the failure threshold for the startup probe
+      failureThreshold: 12
       httpGet:
+        # -- Set the Pod's HTTP path for the startup probe
         path: '{{ default "" .Values.controller.jenkinsUriPrefix }}/login'
+        # -- Set the Pod's HTTP port to use for the startup probe
         port: http
+      # -- Set the time interval between two startup probes executions in seconds
       periodSeconds: 10
+      # -- Set the timeout for the startup probe in seconds
       timeoutSeconds: 5
-      failureThreshold: 12
     livenessProbe:
+      # -- Set the failure threshold for the liveness probe
       failureThreshold: 5
       httpGet:
+        # -- Set the Pod's HTTP path for the liveness probe
         path: '{{ default "" .Values.controller.jenkinsUriPrefix }}/login'
+        # -- Set the Pod's HTTP port to use for the liveness probe
         port: http
+      # -- Set the time interval between two liveness probes executions in seconds
       periodSeconds: 10
+      # -- Set the timeout for the liveness probe in seconds
       timeoutSeconds: 5
+      # -- Set the initial delay for the liveness probe in seconds
       # If Startup Probe is not supported on your Kubernetes cluster, you might want to use "initialDelaySeconds" instead.
       # It delays the initial liveness probe while Jenkins is starting
-      # initialDelaySeconds: 60
+      initialDelaySeconds: 0
```

Comment:
I'm not sure if a null value would be allowed here? Would it automatically use the k8s-default if this value was left empty/null?

Reply:
Looks like tests work with this set to null/empty